x/bank/module.go

```diff
@@ -2,18 +2,27 @@ package bank
 
 import (
 	"context"
+	authmodulev1 "cosmossdk.io/api/cosmos/auth/module/v1"
+	modulev1 "cosmossdk.io/api/cosmos/bank/module/v1"
 	"encoding/json"
 	"fmt"
+	store "github.com/cosmos/cosmos-sdk/store/types"
+	authkeeper "github.com/cosmos/cosmos-sdk/x/auth/keeper"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+
```

Comment:
I don't think we should have direct dependencies here. https://github.com/cosmos/cosmos-sdk/blob/7273bd39e713be8c9dd17bae71b85df495b98be0/x/bank/types/expected_keepers.go#L10 may be what you're looking for

Reply:
that won't work. we need a direct dependency unless we work on #11910

Reply:
would #11910 come before spinning out modules into their own go.mod?

Reply:
it's up to prioritize these things and figure out the ideal design. someone could start on #11910 ASAP if we want to keep the expected keepers pattern

Reply:
#11910 makes sense to me.  I originally used the `AccountKeeper` interface as a parameter in the provider fn but noticed the binding resolution error at start up and reverted to this.  I kind of cringed making this change, but didn't see a way to provide an interface binding in `depinject`.  I was looking for functionality like this [this](https://github.com/cosmos/cosmos-sdk/issues/11910#issuecomment-1142517599) but this could be my Java bias, I think using arbitrary keys to tie things together may be OK too?

Reply:
I don't like the idea of direct module dependencies here tbh.

Reply:
So would you prefer to put the blocked addresses in the bank config then?